core/evaluator_factory.py

```diff
@@ -0,0 +1,189 @@
+#  Drakkar-Software OctoBot
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+from backtesting import backtesting_enabled
+from config import CONFIG_CRYPTO_CURRENCIES, CONFIG_CRYPTO_PAIRS, CONFIG_EVALUATORS_WILDCARD
+from evaluator import TA
+from evaluator.TA import TAEvaluator
+from evaluator.Updaters.global_price_updater import GlobalPriceUpdater
+from evaluator.cryptocurrency_evaluator import CryptocurrencyEvaluator
+from evaluator.evaluator_creator import EvaluatorCreator
+from evaluator.evaluator_task_manager import EvaluatorTaskManager
+from evaluator.symbol_evaluator import SymbolEvaluator
+from services.Dispatchers.dispatcher_creator import DispatcherCreator
+from tools.class_inspector import evaluator_parent_inspection, get_class_from_string
+from tools.logging.logging_util import get_logger
+
+
+class EvaluatorFactory:
+    """EvaluatorFactory class:
+    - Create evaluators
+    """
+
+    def __init__(self, octobot):
+        self.octobot = octobot
+
+        # Logger
+        self.logger = get_logger(self.__class__.__name__)
+
+        self.symbol_tasks_manager = {}
+        self.symbol_evaluator_list = {}
+        self.crypto_currency_evaluator_list = {}
+        self.dispatchers_list = []
+        self.social_eval_tasks = []
+        self.real_time_eval_tasks = []
+
+    def create(self):
+        self.create_dispatchers()
+        self.create_evaluation_tasks()
+
+    def create_dispatchers(self):
+        self.dispatchers_list = DispatcherCreator.create_dispatchers(self.octobot.get_config(), self.octobot.get_async_loop())
+
+    def create_evaluation_tasks(self):
+        self.logger.info("Evaluation threads creation...")
+
+        # create Social and TA evaluators
+        for crypto_currency, crypto_currency_data in self.octobot.get_config()[CONFIG_CRYPTO_CURRENCIES].items():
+            self._create_evaluation_task(crypto_currency, crypto_currency_data)
+
+        self._check_required_evaluators()
+
+    def _create_evaluation_task(self, crypto_currency, crypto_currency_data):
```

Comment:
I thinf it would be clearer to name this method _create_crypto_currency_evaluator_tasks

Reply:
:+1: